SU-SWS/decanter
# ADAPT-2518: Added autorelease github action.

# READY FOR REVIEW

# Summary
Adds autorelease Github action. This action:

- Automatically updates the changelog
- Increments the release version (both in package.json and with a git tag) based on Github label. (i.e. patch, minor, major, alpha, beta, rc)
- Runs 'npm run package' and commits any new changes to dist folder.
- Publishes latest package to npm.

# Needed By (Date)
- Next 2-3 days

# Urgency
- 3/10. (Process improvement, but nothing urgent or time sensitive).

# Steps to Test

This one will be a bit tricky to test, as the action is only fired when merging to "main".
You may need to create a few dummy PRs that make changes to the README file or something along those lines to test the different scenarios.

- Create a new pull request, with base branch of "main".
- Add one of the following labels: "patch", "minor", "major", "alpha", "dev", or "rc"
- Merge the PR.
- Click the "Actions" tab in Github, validate that the action fired correctly and there are no log errors.
- View the "Code" tab in Github. You should see that the CHANGELOG file has been updated with commit messages from the release, package.json updated with a new version number, and a new tag should also be visible.
- Check releases in Github (under code -> tags -> releases) and ensure release notes are present. 
- Github release should also show 'latest release', except for alpha, beta, and rc releases, which should show a "Prerelease" tag.

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/ADAPT-2518
- This is essentially a clone of: https://stanfordits.atlassian.net/browse/ADAPT-2517
- https://www.npmjs.com/package/auto-changelog

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
